src/Connection.cpp

```diff
@@ -76,131 +52,33 @@ struct pollfd Connection::getPollFd() const
-{
-	_headersComplete = headersComplete;
-}
-
-void Connection::setBodyComplete(bool bodyComplete)
-{
-	_bodyComplete = bodyComplete;
-}
-
-void Connection::setHeaders(const std::string &headers)
-{
-	_headers = headers;
-}
-
-void Connection::setBody(const std::string &body)
-{
-	_body = body;
-}
-
-void Connection::setChunkData(const std::string &chunkData)
-{
-	_chunkData = chunkData;
-}
-
-void Connection::setBodyIsChunked(bool bodyIsChunked)
-{
-	_bodyIsChunked = bodyIsChunked;
-}
-
-// Attempts to read HTTP request headers from the client connection into _headers.
-bool Connection::readHeaders()
-{
-	// char buffer[BUFFER_SIZE] = {0};
-	std::cout << "\nreadHeaders" << std::endl;
-	// Both are in the Connection object now
-	// size_t totalRead = 0;
-	// bool headersComplete = false;
-	// while (!headersComplete)
-	if (!_headersComplete)
+	char buffer[BUFFER_SIZE] = {0};
+	ssize_t bytesRead = recv(_pollFd.fd, buffer, BUFFER_SIZE, 0);
+	if (bytesRead > 0)
 	{
-		// We reinitialize it at each iteration to have a clean buffer
-		char buffer[BUFFER_SIZE] = {0};
-		// we could do recv non blocking with MSG_DONTWAIT but we will keep it simple for now
-		std::cout << "_pollFd.fd: " << _pollFd.fd << std::endl;
-		ssize_t bytesRead = recv(_pollFd.fd, buffer, BUFFER_SIZE, 0);
-		std::cout << "bytesRead: " << bytesRead << std::endl;
-		if (bytesRead > 0)
-		{
-			std::cout << "bytesRead > 0" << std::endl;
-			// _headers.append(buffer, bytesRead);
-			_buffer.append(buffer, bytesRead);
-			std::cout << "_buffer: " << _buffer << std::endl;
-			// This will return the index of the first occurrence of "\r\n\r\n" in the string, i.e. the index of '\r' in
-			// the last CRLF
-			std::size_t headersEnd = _buffer.find("\r\n\r\n");
-			if (headersEnd != std::string::npos)
-			{
-				_headers = _buffer.substr(0, headersEnd);
-				_headersComplete = true;
-				// We don't want to include the body in the buffer
-				_buffer = _buffer.substr(headersEnd + 4);
-				_bodyTotalBytesRead = _buffer.size();
-				std::cout << "_headers: " << _headers << std::endl;
-				std::cout << "_buffer: " << _buffer << std::endl;
-				return true;
-			}
-			// std::cout << "_headers: " << _headers << std::endl;
-			_headersTotalBytesRead += bytesRead;
-			if (_headersTotalBytesRead > _clientMaxHeadersSize)
-			{
-				std::cerr << "Header too large" << std::endl;
-				_response.setStatusCode(413);
-				return false;
-			}
-		}
-		else if (bytesRead < 0)
-		{
-			perror("recv failed");
-			return false;
-		}
-		else
-		{
-			// This means biteRead == 0
-			std::cout << "Connection closed before headers being completely sent" << std::endl;
-			// In this case we don't want to send an error response, because the client closed the connection
-			return false;
-		}
+		parser.setBuffer(parser.getBuffer() + std::string(buffer, bytesRead));
+		std::cout << "Exiting readSocket" << std::endl;
+		return true;
+	}
+	else if (bytesRead < 0)
+	{
+		perror("recv failed");
```

Comment:
from the subject:
 _• Checking the value of errno is strictly forbidden after a read or a write operation._

I wonder if  recv is considered "a read operation"...[ I asked the model](https://chat.openai.com/share/6af6870a-6fb9-4e98-8ff9-4498467c30d2) and it answered yes, what do you think ?